olive/hardware/accelerator.py

```diff
@@ -0,0 +1,80 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+from dataclasses import dataclass
+from enum import Enum
+from typing import Union
+
+
+class Device(str, Enum):
+    CPU = "cpu"
+    GPU = "gpu"
+    NPU = "npu"
+    INTEL_MYRIAD = "intel_myriad"
+
+    def __str__(self) -> str:
+        return self.value
+
+
+@dataclass(frozen=True, eq=True)
+class AcceleratorSpec:
+    accelerator_type: Union[str, Device]
+    execution_provider: str
+    vender: str = None
```

Comment:
Are we using these `vender`, `version`, `memory`, `num_cores` for now?

Reply:
from the time being, we don't use them

Reply:
Can we remove this if we are not using this for now? This could simplify the code and will not confuse the user.

Reply:
I add them for future extension